lib/Doctrine/Persistence/Mapping/RuntimeReflectionService.php

```diff
@@ -64,7 +66,9 @@ public function getAccessibleProperty(string $class, string $property)
     {
         $reflectionProperty = new ReflectionProperty($class, $property);
 
-        if ($reflectionProperty->isPublic()) {
+        if (! array_key_exists($property, $this->getClass($class)->getDefaultProperties())) {
```

Comment:
With this change, you can avoid the dependency bump and target 1.3 to fix this bug in a patch release:

```suggestion
        if (class_exists(TypedNoDefaultReflectionProperty::class) && ! array_key_exists($property, $this->getClass($class)->getDefaultProperties())) {
```